test/sequential/test-fs-watch-system-limit.js

```diff
@@ -36,14 +36,27 @@ function spawnProcesses() {
 
 spawnProcesses();
 
+setTimeout(() => {
+  if (!finished) {
+    const timeoutStream = new stream.PassThrough();
+    timeoutStream.write('Error: timeout');
+    timeoutStream.end();
+    timeoutStream.pipe(gatherStderr);
+
+    finished = true;
+    processes.forEach((proc) => proc.kill());
```

Comment:
Is this necessary? Shouldn’t it already be happening below?

Reply:
I was a bit aggressive with the process killing since the test had left my machine with a lot of child processes, but you're right

Reply:
This might be good to add to `process.once('exit')` in case the test fails in the middle.